source/lac/scalapack.cc

```diff
@@ -323,117 +323,298 @@ void ScaLAPACKMatrix<NumberType>::invert()
 
 
 template <typename NumberType>
-std::vector<NumberType> ScaLAPACKMatrix<NumberType>::eigenvalues_symmetric()
+std::vector<NumberType>
+ScaLAPACKMatrix<NumberType>::eigenpairs_symmetric(const bool compute_eigenvectors,
+                                                  const std::pair<int,int> &eigenvalue_idx,
+                                                  const std::pair<NumberType,NumberType> &eigenvalue_limits)
 {
   Assert (state == LAPACKSupport::matrix,
           ExcMessage("Matrix has to be in Matrix state before calling this function."));
   Assert (property == LAPACKSupport::symmetric,
           ExcMessage("Matrix has to be symmetric for this operation."));
+
   Threads::Mutex::ScopedLock lock (mutex);
 
-  ScaLAPACKMatrix<NumberType> Z (grid->n_mpi_processes, grid, 1);
-  std::vector<NumberType> ev (n_rows);
+  // if computation of eigenvectors is not required use a sufficiently small distributed matrix
+  std::unique_ptr<ScaLAPACKMatrix<NumberType>> eigenvectors = compute_eigenvectors ?
+                                                              std::make_unique<ScaLAPACKMatrix<NumberType>>(n_rows, grid, row_block_size)
+                                                              :
+                                                              std::make_unique<ScaLAPACKMatrix<NumberType>>(grid->n_process_rows, grid->n_process_columns,
+                                                                  grid,1,1);
+
+  //ScaLAPACKMatrix<NumberType> eigenvectors (n_rows, grid, row_block_size);
+  eigenvectors->property = property;
+  // number of eigenvalues to be returned; upon successful exit ev contains the m seclected eigenvalues in ascending order
+  int m = n_rows;
+  std::vector<NumberType> ev(n_rows);
```

Comment:
so this is a potentially very big serial vector?

Reply:
Yes, it can not be distributed as ScaLAPACK requires it to be "global".